src/nvim/eval.c

```diff
@@ -16531,7 +16533,7 @@ static void f_termopen(typval_T *argvars, typval_T *rettv, FunPtr fptr)
   curbuf->b_p_swf = false;
   (void)setfname(curbuf, (char_u *)buf, NULL, true);
   // Save the job id and pid in b:terminal_job_{id,pid}
-  Error err;
+  Error err = ERROR_INIT;
   dict_set_var(curbuf->b_vars, cstr_as_string("terminal_job_id"),
                INTEGER_OBJ(rettv->vval.v_number), false, false, &err);
   dict_set_var(curbuf->b_vars, cstr_as_string("terminal_job_pid"),
```

Comment:
Here `err` could be set two times. After the second "set" we would have a memory leak.